[4.0] WebAssetManager: Speed up enable/disable asset

Fedik opened the pull request:
### Summary of Changes

WebAssetManager calculate dependencies on "request" (when it really need) instead of each call of `enableAsset`/`disableAsset`.
This will speed up operations like:
```php
$wa->enableAsset('foo')
  ->enableAsset('bar')
  ->disableAsset('bear')
  ->enableAsset('beer');
```

Additionally I have updated a comment for a Manager interface, added missed throws tags.


### Testing Instructions
Apply the patch and make sure everything still work :wink: 

ping @wilsonge 

for reference #22435

Fedik committed:
WebAssetManager: Speed up enable/disable asset

Fedik commented:
I think I am banned to see ci.joomla.org, so cannot help with Drone, sorry :smile:

dgrammatiko commented:
@Fedik here is the failure:
```
FILE: ...github.com/joomla/joomla-cms/libraries/src/WebAsset/WebAssetManager.php
FOUND 1 ERROR(S) AFFECTING 1 LINE(S)
 91 | ERROR | Expected 1 blank line before member var; 2 found
```

dgrammatiko commented on libraries/src/WebAsset/WebAssetManager.php:
```diff
@@ -80,6 +80,16 @@ class WebAssetManager implements WebAssetManagerInterface, DispatcherAwareInterf
 	 */
 	protected $useVersioning = true;
 
+
```
remove empty line

Fedik replied:
ok now

Fedik committed:
phpcs

Fedik committed:
Remove unnecessary check. Add comment.

wilsonge committed:
Merge branch '4.0-dev' into asset-speed

Fedik committed:
Update a comment for a Manager interface, add missed throws tags

Fedik committed:
Merge branch 'asset-speed' of github.com:Fedik/joomla-cms into asset-speed

Fedik committed:
Sync. Merge branch '4.0-dev' into asset-speed

Fedik committed:
Sync. Merge branch '4.0-dev' into asset-speed

 Conflicts:
	libraries/src/WebAsset/WebAssetManager.php
	libraries/src/WebAsset/WebAssetManagerInterface.php

Fedik committed:
Sync. Merge branch '4.0-dev' into asset-speed

 Conflicts:
	libraries/src/WebAsset/WebAssetManager.php
	libraries/src/WebAsset/WebAssetManagerInterface.php

wilsonge commented:
Thanks!
